File: JavaScript/7js 2/oddOrEven.js
/*
|--------------------------------------------------------------------------
| Odd or Even?
|--------------------------------------------------------------------------
|
*/

function oddOrEven(array) {
   let hsl = 0;
   for (let i in array) {
      hsl += array[i];
   }
   return hsl % 2 == 0 ? "even" : "odd";
}


const oddOrEven = arr => ["even", "odd"][Math.abs(arr.reduce((a, b) => a + b, 0)) % 2];


function oddOrEven(arr) {
   return arr.reduce((a, b) => a + b, 0) % 2 ? 'odd' : 'even';
}


function oddOrEven(array) {
   return array.filter(number => number & 1).length & 1 ? 'odd' : 'even';
}


console.log(oddOrEven([0, 1, 5]), 'even')
console.log(oddOrEven([0, 1, 3]), 'even')
console.log(oddOrEven([1023, 1, 2]), 'even')

File: Python/7py 3/odd_or_even.py
"""
|--------------------------------------------------------------------------
| Odd or Even?
|--------------------------------------------------------------------------
|
"""

def odd_or_even(arr):
   return 'even' if sum(arr) % 2 == 0 else 'odd'


odd_or_even = lambda a: ['even','odd'][sum(a)%2]


def odd_or_even(arr):
   if sum(arr) % 2 == 0:
      return "even"
   else:
      return "odd"


def odd_or_even(arr):
   return ('even', 'odd')[sum(arr) % 2]


print(odd_or_even([0, 1, 2]), "odd")
print(odd_or_even([0, 1, 3]), "even")
print(odd_or_even([1023, 1, 2]), "even")